### ISSUE TITLE
Cloning the repository and using `make` fails

### ISSUE THREAD

**Issue by jsturtevant** [containerd/runwasi#69]
Doing a fresh clone of the repository fails with:

```
 make
cargo build
    Updating crates.io index
   Compiling containerd-shim-wasm v0.1.0 (/home/jstur/projects/runwasi/crates/containerd-shim-wasm)
   Compiling lock_api v0.4.9
   Compiling parking_lot_core v0.9.6
   Compiling wasmedge-sys v0.12.2
   Compiling wasmedge-types v0.3.1
   Compiling oci-tar-builder v0.1.0 (/home/jstur/projects/runwasi/crates/oci-tar-builder)
The following warnings were emitted during compilation:

warning: [wasmedge-sys] Failed to locate lib_dir, include_dir, or header.

error: failed to run custom build command for `wasmedge-sys v0.12.2`

Caused by:
  process didn't exit successfully: `/home/jstur/projects/runwasi/target/debug/build/wasmedge-sys-380b7131222322c7/build-script-build` (exit status: 101)
  --- stdout
  cargo:warning=[wasmedge-sys] Failed to locate lib_dir, include_dir, or header.

  --- stderr
  thread 'main' panicked at '[wasmedge-sys] Failed to locate the required header and/or library file. Please reference the link: https://wasmedge.org/book/en/embed/rust.html', /home/jstur/.cargo/registry/src/github.com-1ecc6299db9ec823/wasmedge-sys-0.12.2/build.rs:30:25
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
make: *** [Makefile:19: build] Error 101
```

Should `wasmedge` be behind a feature flag in the crate or should we add something to the `makefile` to install the correct dependencies?

**Comment by devigned** [comment]
Another option would be a build container or use cross-rs.

**Comment by jsturtevant** [comment]
After installing wasmedge with `curl -sSf https://raw.githubusercontent.com/WasmEdge/WasmEdge/master/utils/install.sh | bash`

I am getting: 

```
 make
cargo build
   Compiling wasmedge-sys v0.12.2
The following warnings were emitted during compilation:

warning: [wasmedge-sys] libwasmedge found in /home/jstur/.wasmedge/lib

error: failed to run custom build command for `wasmedge-sys v0.12.2`

Caused by:
  process didn't exit successfully: `/home/jstur/projects/runwasi/target/debug/build/wasmedge-sys-380b7131222322c7/build-script-build` (exit status: 101)
  --- stdout
  cargo:warning=[wasmedge-sys] libwasmedge found in /home/jstur/.wasmedge/lib

  --- stderr
  thread 'main' panicked at 'Unable to find libclang: "couldn't find any valid shared libraries matching: ['libclang.so', 'libclang-*.so', 'libclang.so.*', 'libclang-*.so.*'], set the `LIBCLANG_PATH` environment variable to a path where one of these files can be found (invalid: [])"', /home/jstur/.cargo/registry/src/github.com-1ecc6299db9ec823/bindgen-0.61.0/./lib.rs:2276:31
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
make: *** [Makefile:19: build] Error 101
```

**Comment by jsturtevant** [comment]
was able to compile after `sudo apt-get -y install libclang-15-dev`

**Comment by jsturtevant** [comment]
found docs on the readme:

```
$ sudo -E sh -c 'echo "$HOME/.wasmedge/lib" > /etc/ld.so.conf.d/libwasmedge.conf'
$ sudo ldconfig
```
and tests pass now

**Comment by jsturtevant** [comment]
Ok, so there is a `make` target that helps with install and setup: https://github.com/containerd/runwasi/blob/a2f86e43e8df4efece9ec4b3f9460421f450c8fc/Makefile#L63-L65

**Comment by defims** [comment]
> Ok, so there is a `make` target that helps with install and setup:
> 
> https://github.com/containerd/runwasi/blob/a2f86e43e8df4efece9ec4b3f9460421f450c8fc/Makefile#L63-L65

You may also need to set enviroment varibles after `make bin/wamsedge`:

```
export WASMEDGE_INCLUDE_DIR=$(PWD)/bin/wasmedge/include && \
export WASMEDGE_LIB_DIR=$(PWD)/bin/wasmedge/lib
```

**Comment by denis2glez** [comment]
It will probably be fixed by #103 if you followed the README file instructions to install WasmEdge.

**Comment by Mossaka** [comment]
@jsturtevant I think we can close this one since a lot has changed.
